engine/access/rpc/backend/backend_transactions_test.go

```diff
@@ -0,0 +1,329 @@
+		tx := unittest.TransactionFixture()
+		tx.TransactionBody = tbody
+
+		coll := flow.CollectionFromTransactions([]*flow.Transaction{&tx})
+
+		suite.transactions.
+			On("ByID", tx.ID()).
+			Return(nil, storage.ErrNotFound)
+
+		suite.blocks.
+			On("ByID", block.ID()).
+			Return(&block, nil).
+			Once()
+
+		suite.communicator.On("CallAvailableNode",
+			mock.Anything,
+			mock.Anything,
+			mock.Anything).
+			Return(nil)
+
+		receipt := unittest.ExecutionReceiptFixture()
+		identity := unittest.IdentityFixture()
+		identity.Role = flow.RoleExecution
+
+		suite.state.On("AtBlockID", block.ID()).Return(snap, nil).Once()
+
+		l := flow.ExecutionReceiptList{receipt}
+
+		transactionResultResponse := access.TransactionResultResponse{
+			Status:     entities.TransactionStatus_EXECUTED,
+			StatusCode: uint32(entities.TransactionStatus_EXECUTED),
+		}
+
+		suite.receipts.
+			On("ByBlockID", block.ID()).
+			Return(l, nil)
+
+		suite.historicalAccessClient.
+			On("GetTransactionResult", mock.Anything, mock.Anything).
+			Return(&transactionResultResponse, nil)
+
+		backend := New(
+			suite.state,
+			suite.colClient,
+			[]access.AccessAPIClient{suite.historicalAccessClient},
+			suite.blocks,
+			nil,
+			nil,
+			suite.transactions,
+			suite.receipts,
+			nil,
+			suite.chainID,
+			metrics.NewNoopCollector(),
+			nil,
+			false,
+			DefaultMaxHeightRange,
+			nil,
+			nil,
+			suite.log,
+			DefaultSnapshotHistoryLimit,
+			nil,
+			*suite.communicator,
+			false,
+		)
+		resp, err := backend.GetTransactionResult(context.Background(), tx.ID(), block.ID(), coll.ID())
+		suite.Require().NoError(err)
+		suite.Require().Equal(flow.TransactionStatusExecuted, resp.Status)
+		suite.Require().Equal(uint(flow.TransactionStatusExecuted), resp.StatusCode)
+	})
+}
+
+func (suite *Suite) TestGetTransactionResultFromCache() {
+	suite.WithPreConfiguredState(func(snap protocol.Snapshot) {
+		block := unittest.BlockFixture()
+		tbody := unittest.TransactionBodyFixture()
+		tx := unittest.TransactionFixture()
+		tx.TransactionBody = tbody
+
+		coll := flow.CollectionFromTransactions([]*flow.Transaction{&tx})
+
+		suite.transactions.
+			On("ByID", tx.ID()).
+			Return(nil, storage.ErrNotFound)
+
+		suite.blocks.
+			On("ByID", block.ID()).
+			Return(&block, nil).
+			Once()
+
+		suite.communicator.On("CallAvailableNode",
+			mock.Anything,
+			mock.Anything,
+			mock.Anything).
+			Return(nil)
+
+		receipt := unittest.ExecutionReceiptFixture()
+		identity := unittest.IdentityFixture()
+		identity.Role = flow.RoleExecution
+
+		suite.state.On("AtBlockID", block.ID()).Return(snap, nil).Once()
+
+		l := flow.ExecutionReceiptList{receipt}
+
+		transactionResultResponse := access.TransactionResultResponse{
+			Status:     entities.TransactionStatus_EXECUTED,
+			StatusCode: uint32(entities.TransactionStatus_EXECUTED),
+		}
+
+		suite.receipts.
+			On("ByBlockID", block.ID()).
+			Return(l, nil)
+
+		suite.historicalAccessClient.
```

Comment:
```suggestion
		// assert this is called exactly once for the 2 API calls
		suite.historicalAccessClient.
```

Reply:
https://github.com/onflow/flow-go/pull/4598/commits/f018aca1c2d726305cfd9ee91a9206858c18eb0a

Reply:
can you also add the comment. it makes it easier for the next person to look at these tests to quickly understand what's happening